test/e2e/kind_test.go

```diff
@@ -202,23 +224,123 @@ func (s *kindSuite) TestKindRun() {
+		s.EventuallyWithTf(func(c *assert.CollectT) {
+			clusterAgentPods, err := k8s.ListPodsE(t, kubectlOptions, v1.ListOptions{
+				LabelSelector: clusterAgentSelector + ",agent.datadoghq.com/e2e-test=datadog-agent-minimum",
+			})
+			assert.NoError(t, err)
+
+			for _, pod := range clusterAgentPods {
+				k8s.WaitUntilPodAvailable(t, kubectlOptions, pod.Name, 9, 15*time.Second)
+				output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", "-it", pod.Name, "--", "agent", "status", "collector", "-j")
+				assert.NoError(t, err)
+
+				verifyCheck(c, output, "kubernetes_state_core")
+			}
+		}, 900*time.Second, 30*time.Second, "could not validate kubernetes_state_core check on cluster agent pod")
 
 		s.EventuallyWithTf(func(c *assert.CollectT) {
-			now = time.Now()
-			series, err := s.datadogClient.QueryMetrics(now.Add(-1*time.Minute).Unix(), now.Unix(), metricQuery)
-			assert.Truef(c, len(series) > 0, "expected metric series to not be empty: %s", err)
-		}, 600*time.Second, 30*time.Second, "metric series has not changed to not empty")
+			verifyKSMCheck(s, c)
+		}, 600*time.Second, 30*time.Second, "could not validate kubernetes_state_core check with api client")
+	})
+
+	s.T().Run("KSM Check Works (cluster check runner)", func(t *testing.T) {
+		// Update DDA
+		ddaConfigPath, err = getAbsPath(filepath.Join(manifestsPath, "datadog-agent-ccr-enabled.yaml"))
+		assert.NoError(t, err)
+		k8s.KubectlApply(t, kubectlOptions, ddaConfigPath)
+		verifyAgentPods(t, kubectlOptions, nodeAgentSelector+",agent.datadoghq.com/e2e-test=datadog-agent-ccr-enabled")
+		verifyNumPodsForSelector(t, kubectlOptions, 1, clusterAgentSelector+",agent.datadoghq.com/e2e-test=datadog-agent-ccr-enabled")
+		verifyNumPodsForSelector(t, kubectlOptions, 1, clusterCheckRunnerSelector+",agent.datadoghq.com/e2e-test=datadog-agent-ccr-enabled")
+
+		s.EventuallyWithTf(func(c *assert.CollectT) {
+			ccrPods, err := k8s.ListPodsE(t, kubectlOptions, v1.ListOptions{
+				LabelSelector: clusterCheckRunnerSelector + ",agent.datadoghq.com/e2e-test=datadog-agent-ccr-enabled",
+			})
+			assert.NoError(c, err)
+
+			for _, ccr := range ccrPods {
+				k8s.WaitUntilPodAvailable(t, kubectlOptions, ccr.Name, 9, 15*time.Second)
+				output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "exec", "-it", ccr.Name, "--", "agent", "status", "collector", "-j")
+				assert.NoError(c, err)
+
+				verifyCheck(c, output, "kubernetes_state_core")
+			}
+		}, 900*time.Second, 30*time.Second, "could not validate kubernetes_state_core check on cluster check runners pod")
+
+		s.EventuallyWithTf(func(c *assert.CollectT) {
+			verifyKSMCheck(s, c)
+		}, 600*time.Second, 30*time.Second, "could not validate kubernetes_state_core check with api client")
 	})
 
 	s.T().Run("Cleanup DDA", func(t *testing.T) {
 		deleteDda(t, kubectlOptions, ddaConfigPath)
 	})
 }
+func verifyCheck(c *assert.CollectT, collectorOutput string, checkName string) {
```

Comment:
Nice to have a common function for verifying checks, such primitives will simplify later test!